docs/security.md

```diff
@@ -0,0 +1,106 @@
+# Creating Secure Clusters
+
+## Enabling Kerberos
+
+To enable Kerberos for a cluster, set `kerberos: true` inside the `security` block of its definition.
+Kerberos **must** be enabled if you want to include certain services in the cluster, e.g. Ranger or Sentry.
+
+```yaml
+clusters:
+  - name: CDP PvC Base Cluster
+    services: [HDFS, RANGER, SOLR, ZOOKEEPER]
+    security:
+      kerberos: true
+```
+
+If a KDC will be provisioned by the playbook and you are happy with default values for realm names, encryption types etc, no other configuration is required.
+
+## Enabling TLS
+
+To enable TLS for a cluster, the following configurations are required:
+
+1) Set `tls: true` inside the `security` block of its definition.
+
+```yaml
+clusters:
+  - name: CDP PvC Base Cluster
+    services: [HDFS, RANGER, SOLR, ZOOKEEPER]
+    security:
+      tls: true
+```
+
+2) Set `tls: True` on all inventory host groups for which TLS keystores and truststores need to be generated.
+
+```ini
+[cluster:vars]
```

Comment:
I believe this should actually be [deployment:vars] as you need the certs on DB servers and other things as well?

Reply:
Fixed